src/ray/raylet/node_manager.cc

```diff
@@ -312,6 +314,12 @@ void NodeManager::Heartbeat() {
     last_debug_dump_at_ms_ = now_ms;
   }
 
+  // Evict all copies of freed objects from the cluster.
+  if (free_objects_period_ > 0 &&
```

Comment:
Doing this in the heartbeat means the flush duration needs to be much larger than or a multiple of the heartbeat duration - might be better to just have a separate timer on the event loop. Otherwise should indicate this in the comments in the config.